effdet/loss.py

```diff
@@ -182,12 +219,29 @@ def forward(
             cls_loss = cls_loss.view(bs, height, width, -1, self.num_classes)
             cls_loss *= (cls_targets_at_level != -2).unsqueeze(-1).float()
             cls_losses.append(cls_loss.sum())
+            if not self.use_iou_loss:
+                box_losses.append(_box_loss(
+                    box_outputs[l].permute(0, 2, 3, 1),
+                    box_targets_at_level,
+                    num_positives_sum,
+                    delta=self.delta))
+
+            else:
+                box_outputs_list.append(box_outputs[l].permute(0, 2, 3, 1).reshape([bs, -1, 4]))
+                cls_targets_list.append(cls_targets_at_level.permute(0, 2, 3, 1).reshape([bs, -1, 1]))
+                box_targets_list.append(box_targets_at_level.permute(0, 2, 3, 1).reshape([bs, -1, 4]))
+
+
+        if self.use_iou_loss:
+            # apply bounding box regression to anchors
+            for k in range(box_outputs_list.shape[0]):
+                pred_boxes = decode_box_outputs(box_outputs_list[k].T.float(), self.anchors.boxes.T, output_xyxy=True)
```

Comment:
It also a bug,     ty, tx, th, tw = rel_codes.unbind(dim=1)
ValueError: too many values to unpack (expected 4)

Reply:
I will check it, I think I was not working with the repo's last commit

Reply:
thanks for your reply,  looking forward your new correct code

Reply:
Do you mind share your variable giou loss with effdet version?